app/charts/column/columns-stacked-state.tsx

```diff
@@ -52,12 +52,12 @@ import {
 } from "@/charts/shared/stacked-helpers";
 import useChartFormatters from "@/charts/shared/use-chart-formatters";
 import { InteractionProvider } from "@/charts/shared/use-interaction";
-import { useInteractiveFilters } from "@/charts/shared/use-interactive-filters";
 import { Observer, useWidth } from "@/charts/shared/use-width";
 import { ColumnConfig } from "@/configurator";
 import { Observation } from "@/domain/data";
 import { useFormatNumber } from "@/formatters";
 import { getPalette } from "@/palettes";
+import { useInteractiveFiltersStore } from "@/stores/interactive-filters";
```

Comment:
My main concern concerning using a zustand store is that it is global, meaning that if I am not mistaken, it will be problematic to display 2 charts on the same page.

Reply:
Yes, but we'll be able to have something like `createInteractiveFiltersStore` function, and that store could be then created within each chart (and there might be a case, where they should be global, as will be the case for dashboards) 👀

Reply:
Yes 👍  Then you will need a provider around the chart to provide the created store right ? In the end it will not be that different than with the provider right ? Do you see an advantage ? Not saying that we should change it back, jsut wanting to know that there is advantage I am not seeing.

Reply:
Yes, we would need a provider then (or change the logic to pass the object down to the components). I used Zustand to be able to use `getState()` method which works outside of the React component context (and allowed to use up-to-date state in the brush callback). Other way of solving it would be to create a "global" variable getter, but I wanted to avoid that 👀

Reply:
Also, with atomic state picks we can only re-render the components when a given value changes (and I believe in the old way, we were re-rendering a component when any value changed inside the reducer) 🤔